addons/website/models/website.py

```diff
@@ -932,10 +941,29 @@ def get_current_website(self, fallback=True):
 
         # The format of `httprequest.host` is `domain:port`
         domain_name = request and request.httprequest.host or ''
+        if is_frontend_request:
+            website_id = self._get_current_website_id(domain_name, fallback=fallback)
+        elif fallback:
+            # We are in a backend context, or early in the dispatch stack and
+            # a fallback website is requested. Before returning a random one,
+            # let's try to return the one matching the request host/domain if
+            # possible.
+            if domain_name:
+                website_id = self._get_current_website_id(domain_name, fallback=fallback)
+            else:
+                website_id = self._get_first_website_id()
```

Comment:
Not entirely sure the "else" part is a needed change:
- If we have a domain, we best use it -> nice
- If we don't have one... should we care the one is returned? This could be let to _get_current_website_id ? :shrug: